### PR TITLE
[FLINK-14723][table-planner-blink] Improve some code of computed column in planner

### PR DESCRIPTION
**Pull request by wuchong**
## What is the purpose of the change

This pull request improves some code of computed column. 

## Brief change log

1. `FlinkRelOptUtilTest#testToString` shouldn't ignore attributes of RelNodes. The original expected string makes sense. We should fix it to keep the result unchanged.
2. The test `CatalogTableITCase#testInsertSinkTableWithUnmatchedFields` should fail and shouldn't verify the expected results (no results).
3. `SqlToOperationConverterTest#testCreateTableWithComputedColumn` change the udf calling to non-qualified way to verify the qualifying works well.

## Verifying this change

This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by wuchong** [comment]
cc @danny0405

File: flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/operations/JavaDataStreamQueryOperation.java [hunk 1]
```diff
@@ -38,6 +38,11 @@
 @Internal
 public class JavaDataStreamQueryOperation<E> implements QueryOperation {
 
+	// JavaDataStreamQueryOperation represent a registered DataStream from `registerDataStream` method
+	// or a unregistered DataStream from `fromDataStream`. However the name under which the DataStream
+	// is registered is missing when converting a JavaDataStreamQueryOperation to a RelNode.
+	// So using `qualifiedName` to keep the original name.
+	private List<String> qualifiedName;
```
Comment (KurtYoung) [inline flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/operations/JavaDataStreamQueryOperation.java]: Why not storing it in TableSchema?

File: flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/operations/SqlToOperationConverterTest.java [hunk 2]
```diff
@@ -390,14 +390,14 @@ public void testCreateTableWithComputedColumn() {
 			"  b varchar, \n" +
 			"  c as a - 1, \n" +
 			"  d as b || '$$', \n" +
-			"  e as my_catalog.my_database.my_udf(a)\n" +
+			"  e as my_udf(a)\n" +
```
Comment (KurtYoung) [inline flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/operations/SqlToOperationConverterTest.java]: We should cover 3 scenarios:
1. my_udf
2. db.udf1
3. catalog.db.udf2

File: flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/operations/JavaDataStreamQueryOperation.java (hunk 1 above)
Comment (wuchong) [reply flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/operations/JavaDataStreamQueryOperation.java]: I don't fully understand. How to store this in `TableSchema`?

**Comment by KurtYoung** [comment]
btw, you should make the jira title more specific.

File: flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/operations/JavaDataStreamQueryOperation.java (hunk 1 above)
Comment (KurtYoung) [reply flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/operations/JavaDataStreamQueryOperation.java]: I misunderstood this as field names..

File: flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/operations/JavaDataStreamQueryOperation.java [hunk 3]
```diff
@@ -38,6 +38,11 @@
 @Internal
 public class JavaDataStreamQueryOperation<E> implements QueryOperation {
 
+	// JavaDataStreamQueryOperation represent a registered DataStream from `registerDataStream` method
```
Comment (KurtYoung) [inline flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/operations/JavaDataStreamQueryOperation.java]: These seems more suitable to be commit message but not as comments.

**Comment by wuchong** [comment]
Thanks @KurtYoung for the reviewing. I splitted the commit into 3 separate commits with more specific commit messages.

**Comment by wuchong** [comment]
Thanks @KurtYoung for the reviewing. I splitted the commit into 3 separate commits with more specific commit messages.

File: flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/operations/JavaDataStreamQueryOperation.java (hunk 1 above)
Comment (dawidwys) [reply flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/operations/JavaDataStreamQueryOperation.java]: Can we use ObjectIdentifier here?

File: flink-table/flink-table-api-scala-bridge/src/main/java/org/apache/flink/table/operations/ScalaDataStreamQueryOperation.java [hunk 4]
```diff
@@ -38,6 +38,11 @@
 @Internal
 public class ScalaDataStreamQueryOperation<E> implements QueryOperation {
 
+	// ScalaDataStreamQueryOperation represent a registered DataStream from `registerDataStream` method
+	// or an unregistered DataStream from `fromDataStream`. However the name under which the DataStream
+	// is registered is missing when converting a ScalaDataStreamQueryOperation to a RelNode.
+	// So using `qualifiedName` to keep the original name.
+	private List<String> qualifiedName;
```
Comment (dawidwys) [inline flink-table/flink-table-api-scala-bridge/src/main/java/org/apache/flink/table/operations/ScalaDataStreamQueryOperation.java]: Can we use ObjectIdentifier here?

**Comment by wuchong** [comment]
Thanks @dawidwys , I refactored it to use `ObjectIdentifier`.

File: flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/operations/JavaDataStreamQueryOperation.java (hunk 3 above)
Comment (KurtYoung) [reply flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/operations/JavaDataStreamQueryOperation.java]: what do you think about this?

Comment (wuchong) [reply flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/operations/JavaDataStreamQueryOperation.java]: I have added this to commit message. But also keep is as comments.

Comment (KurtYoung) [reply flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/operations/JavaDataStreamQueryOperation.java]: I don't get why we want to keep this as comments? It's just a table name, and due to some reason we lost it in the old code, but added back now. We don't have to record all the reasons behind this in comment.

Comment (wuchong) [reply flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/operations/JavaDataStreamQueryOperation.java]: I think it explains clearly why we have this field in it (we don't have this information when initialization but pass in when converting to RelNode). 

But I'm also fine to remove this.

File: flink-table/flink-table-api-scala-bridge/src/main/java/org/apache/flink/table/operations/ScalaDataStreamQueryOperation.java [hunk 5]
```diff
@@ -38,6 +39,7 @@
 @Internal
 public class ScalaDataStreamQueryOperation<E> implements QueryOperation {
 
+	private ObjectIdentifier identifier;
```
Comment (KurtYoung) [inline flink-table/flink-table-api-scala-bridge/src/main/java/org/apache/flink/table/operations/ScalaDataStreamQueryOperation.java]: Is it possible to have this during construction?

Comment (wuchong) [reply flink-table/flink-table-api-scala-bridge/src/main/java/org/apache/flink/table/operations/ScalaDataStreamQueryOperation.java]: We don't know the registered table name when construct this instance from `fromDataStream(DataStream<T> dataStream)`.

Comment (KurtYoung) [reply flink-table/flink-table-api-scala-bridge/src/main/java/org/apache/flink/table/operations/ScalaDataStreamQueryOperation.java]: So when the name is decided?

**Pull request closed without merging**